tools/runfiles/src/Bazel/Runfiles.hs

```diff
@@ -18,32 +20,52 @@ import Control.Monad.Trans.Maybe (MaybeT (..))
 import Control.Monad.IO.Class (liftIO)
 import Data.Char (toLower)
 import Data.Foldable (asum)
-import Data.List (find, isPrefixOf, isSuffixOf)
+import Data.List (find, isPrefixOf, isSuffixOf, intercalate)
 import Data.Maybe (fromMaybe)
 import GHC.Stack
 import System.Directory (doesDirectoryExist, doesFileExist, listDirectory, makeAbsolute)
 import System.Environment (lookupEnv)
 import qualified System.FilePath
-import System.FilePath ((</>), (<.>), addTrailingPathSeparator, takeFileName)
+import System.FilePath ((</>), (<.>), addTrailingPathSeparator, takeFileName, splitDirectories, isAbsolute)
 import System.Info (os)
 
+-- | Bazel repository mapping for bzlmod
+-- See https://github.com/bazelbuild/proposals/blob/7c8da4a931d83db5c25abf85f6c486ad22d330e3/designs/2022-07-21-locating-runfiles-with-bzlmod.md
+type RepoMapping = [((String, String), String)]
```

Comment:
According to [this file](https://github.com/commercialhaskell/stackage-content/blob/master/stack/global-hints.yaml) it seems that `text` is only a core package since `ghc-8.4`, but `containers` is there for all the versions mentioned in the file (which start at `ghc-7.10.1`).

I updated the library to use `Data.Map.Strict` in [this commit](https://github.com/tweag/rules_haskell/pull/1912/commits/8748511b9012062798a83c0474a1b6e63f0639d7) (the `lookupDir` function makes use of `Map.to_list` but I think it's fine since we need to iterate over (key,value) pairs in this case).